tests/backup_e2e.rs

```diff
@@ -0,0 +1,27 @@
+// This is an integration test file that lives in the tests/ directory
+// Integration tests verify that multiple components work together correctly
+mod common;  // Helper module for shared test utilities
+
+use common::{create_test_folder};  // Import test utilities
+use std::process::Command;  // For executing CLI commands
+
+#[test]
+fn test_backup_via_cli() -> Result<(), Box<dyn std::error::Error>> {
+    // Create temporary test directories using helper from common/mod.rs
+    let source_dir = create_test_folder();
+    let dest_dir = create_test_folder();
+    
+    // Test the CLI by actually running the binary with cargo run
+    // This tests the full application stack from CLI to core logic
+    let status = Command::new("cargo")
```

Comment:
This doesn't follow the pattern used in gitopolis, is there a reason for the alternative approach? I think I prefer the gitopolis approach because it tests the compiled binary with its real arguments, rather than testing via cargo which is irrelevant to the operation of the final program. Adding cargo to the test will also make it fragile to changes in the cli of cargo, which is undesirable

https://github.com/rustworkshop/gitopolis/blob/989381071fac589fc4f41f4da03ffde9320a3213/tests/end_to_end_tests.rs#L771